ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -308,110 +255,6 @@ public FetchTask getFetchTask() {
-          }
-        }
-      }
-
-      if (driverContext.getPlan().getAcidAnalyzeTable() != null) {
-        // Allocate write ID for the table being analyzed.
-        Table t = driverContext.getPlan().getAcidAnalyzeTable().getTable();
-        driverContext.getTxnManager().getTableWriteId(t.getDbName(), t.getTableName());
-      }
-
-
-      DDLDescWithWriteId acidDdlDesc = driverContext.getPlan().getAcidDdlDesc();
-      boolean hasAcidDdl = acidDdlDesc != null && acidDdlDesc.mayNeedWriteId();
-      if (hasAcidDdl) {
-        String fqTableName = acidDdlDesc.getFullTableName();
-        final TableName tn = HiveTableName.ofNullableWithNoDefault(fqTableName);
-        long writeId = driverContext.getTxnManager().getTableWriteId(tn.getDb(), tn.getTable());
-        acidDdlDesc.setWriteId(writeId);
-      }
-
-      /*It's imperative that {@code acquireLocks()} is called for all commands so that
-      HiveTxnManager can transition its state machine correctly*/
-      driverContext.getTxnManager().acquireLocks(driverContext.getPlan(), context, userFromUGI, driverState);
-      final List<HiveLock> locks = context.getHiveLocks();
-      LOG.info("Operation {} obtained {} locks", driverContext.getPlan().getOperation(),
-          ((locks == null) ? 0 : locks.size()));
-      // This check is for controlling the correctness of the current state
-      if (driverContext.getTxnManager().recordSnapshot(driverContext.getPlan()) &&
-          !driverContext.isValidTxnListsGenerated()) {
-        throw new IllegalStateException(
-            "Need to record valid WriteID list but there is no valid TxnID list (" +
-                JavaUtils.txnIdToString(driverContext.getTxnManager().getCurrentTxnId()) +
-                ", queryId:" + driverContext.getPlan().getQueryId() + ")");
-      }
-
-      if (driverContext.getPlan().hasAcidResourcesInQuery() || hasAcidDdl) {
-        validTxnManager.recordValidWriteIds();
-      }
-
-    } catch (Exception e) {
-      String errorMessage;
-      if (driverState.isDestroyed() || driverState.isAborted() || driverState.isClosed()) {
-        errorMessage = String.format("Ignore lock acquisition related exception in terminal state (%s): %s",
-            driverState.toString(), e.getMessage());
-        CONSOLE.printInfo(errorMessage);
-      } else {
-        errorMessage = String.format("FAILED: Error in acquiring locks: %s", e.getMessage());
-        CONSOLE.printError(errorMessage, "\n" + StringUtils.stringifyException(e));
-      }
-      throw DriverUtils.createProcessorException(driverContext, 10, errorMessage, ErrorMsg.findSQLState(e.getMessage()),
-          e);
-    } finally {
-      perfLogger.PerfLogEnd(CLASS_NAME, PerfLogger.ACQUIRE_READ_WRITE_LOCKS);
-    }
-  }
-
   public void releaseLocksAndCommitOrRollback(boolean commit) throws LockException {
     releaseLocksAndCommitOrRollback(commit, driverContext.getTxnManager());
   }
```

Comment:
What about removing Driver.releaseLocksAndCommitOrRollback altogether, and call driverTxnHandler.releaseLocksAndCommitOrRollback(commit, txnManager) immediately?

Reply:
I'm planning to remove both releaseLocksAndCommitOrRollback functions from Driver, but unfortunately they are both used by some external callers. In a future patch I'll try to remove those.

Reply:
Sounds good